### PR TITLE
add getDate() method for #80

### LINKED ISSUE #80: [FEATURE] - Add `date` get.
**Description**

To implement v3.1 specification, this package needs to expose a `getDate` property which returns the number as a date (year, month, date) in a suitable structure.

**Breaking changes**

No breaking changes expected.

### PR DESCRIPTION
**Pull request by cbhat5**
…m the Personnummer object

**Type of change**

- [x] New feature
- [ ] Bug fix
- [ ] Security patch
- [ ] Documentation update

**Description**

 
Add the getDate method for #80 

**Related issue**
 Closes #80 
**Motivation**

All good.

**Checklist**

- [x] I have read the **CONTRIBUTING** document.
- [x] I have read and accepted the **Code of conduct**
- [x] Tests passes.
- [x] Style lints passes.
- [x] Documentation of new public methods exists.

### CODE REVIEW

**Comment by cbhat5** [comment]
Might have to change if #60 is merged.

**Review by cbhat5 — COMMENTED** [review]
Will have to change getDate() month component if #60 is merged

File: src/test/java/PersonnummerTest.java [hunk 1]
```diff
@@ -60,6 +60,15 @@ public void testParse(PersonnummerData ssn) {
         assertDoesNotThrow(() -> Personnummer.parse(ssn.separatedFormat, new Options(false)));
         assertDoesNotThrow(() -> Personnummer.parse(ssn.separatedFormat, new Options(false)));
     }
+    
+    @ParameterizedTest
+    @MethodSource("DataProvider#getDate")
+    public void testDate(PersonnummerData ssn) {
```
Comment (Johannestegner) [inline src/test/java/PersonnummerTest.java:66]: Could you add a test on the format as well just to make sure the actual dates are correct (basically just check that the passed SSN have the correct year/month/day compared to the actual date object)?

**Review by Johannestegner — CHANGES_REQUESTED** [review]
Thank you for the pull request!

The code looks good, I'd just love a check on the actual output in the tests and I'll happily merge it :)

**Comment by cbhat5** [comment]
@Johannestegner The existing code with below test case sets the `fullYear` to be `2009` instead of `1909` for `separated_format`/`short format`. I create `LocalDateTime`object based on `fullYear` attribute. The base code needs to be rewritten to consider the space.
```json
{
        "integer": 0,
        "long_format": "19090527 1474",
        "short_format": "090527 1474",
        "separated_format": "090527 1474",
        "separated_long": "19090527 1474",
        "valid": false,
        "type": "ssn",
        "isMale": true,
        "isFemale": false
    }
```

**Comment by Johannestegner** [comment]
When it comes to the short format, we always have to guess which century the number is for, so that one is expected.
When it comes to the separated format, if there is a `-`, the number will be seen as within the 100 year range, and a `+` over 100 years.

So the full year should be correct to use.

Easiest way to handle this in the tests would be to just check on the decade value instead of the full year in the assert :)

**Comment by Johannestegner** [comment]
In the C# package, I format the value from the Date object as a `yyMMdd` value and just check them against each other, which I think is a sufficient test.

**Comment by cbhat5** [comment]
> When it comes to the short format, we always have to guess which century the number is for, so that one is expected. When it comes to the separated format, if there is a `-`, the number will be seen as within the 100 year range, and a `+` over 100 years.
> 
> So the full year should be correct to use.
> 
> Easiest way to handle this in the tests would be to just check on the decade value instead of the full year in the assert :)

Will not the short format also have a separator either of `-` or `+` instead of the ` ` in the short format?

File: src/test/java/PersonnummerTest.java [hunk 2]
```diff
@@ -60,6 +62,33 @@ public void testParse(PersonnummerData ssn) {
...
+			data.add(new Personnummer(ssn.separatedFormat, new Options()));
+			
+			data.forEach(entry -> {
+				assertDoesNotThrow(() -> entry.getDate());
+				
+				String expected = entry.getFullYear();
+//				Integer expected = Integer.valueOf(ssn.longFormat.substring(0,4));
+				String actual = String.valueOf(entry.getDate().getYear());
+				
+				assertEquals(expected, actual, "expected = " + expected + "\n" + "Actual = " + actual);
```
Comment (Johannestegner) [inline src/test/java/PersonnummerTest.java:82]: I was thinking checking all the four parts here, YY MM and DD.
Just against a simple format would work fine:

```
// something like...
assertEquals(ssn.longFormat.substring(2),  date.getYear() + date.getMonth() + date.getDay());
```

Although, I'd guess that the date object have some type of format function with a string value to pass (yyMMdd would do in that case I would imagine).

With that I'll happily merge the PR :)

**Comment by cbhat5** [comment]
> Aaah I double checked the meta (and then again looked at the code you shown) and that number is invalid (due to the space), so it should fail in the parse even :)

yeah, looks like an issue with regex, I added a space as an additional option, and it fails many other tests, I am not sure of the other possible patterns. Does the updated PR look fine to you? It passes the tests with `getFullYear`.

File: src/test/java/PersonnummerTest.java (hunk 2 above)
Comment (cbhat5) [reply src/test/java/PersonnummerTest.java:82]: Ok, I'll add the dates and months. I totally forgot with this century issue popping up.

**Review by Johannestegner — APPROVED** [review]

**Comment by Johannestegner** [comment]
Yes, looks good to me! :)

Thank you for the pull request!

**Pull request merged**